plugins/rbac/src/components/DeleteRoleDialog.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+
+import { ErrorBoundary } from '@backstage/core-components';
+import { useApi } from '@backstage/core-plugin-api';
+
+import {
+  Box,
+  Button,
+  createStyles,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  makeStyles,
+  Theme,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+
+import { rbacApiRef } from '../api/RBACBackendClient';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    titleContainer: {
+      display: 'flex',
+      alignItems: 'center',
+      gap: theme.spacing(1),
+    },
+    closeButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+  }),
+);
+
+type DeleteRoleDialogProps = {
+  open: boolean;
+  closeDialog: () => void;
+  roleName: string;
+};
+const DeleteRoleDialog = ({
+  open,
+  closeDialog,
+  roleName,
+}: DeleteRoleDialogProps) => {
+  const classes = useStyles();
+
+  const rbacApi = useApi(rbacApiRef);
+
+  const deleteRole = async () => {
+    try {
+      await rbacApi.deleteRole(roleName);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+    }
```

Comment:
@debsmita1 If an error occurs while deleting a role, IMO we should notify the users that there was some error while deleting the role and they can try again or something similar.

cc: @invincibleJai @parvathyvr